src/test/java/pages/WomenTopsPage.java

```diff
@@ -0,0 +1,68 @@
+package pages;
+
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class WomenTopsPage extends AbstractPage {
+
+    private final By colorFilterLocator = By.xpath("//div[@class='filter-options-title' and contains(text(), 'Color')]");
+    private final By selectedColorLocator = By.cssSelector("[aria-label='Color'] [option-label='Blue']");
+    private final By priceFilterLocator = By.xpath("//div[@class='filter-options-title' and contains(text(), 'Price')]");
+    private final By clearFilterLocator = By.cssSelector(".filter-clear");
+    private final By filteredItemsPriceLocator = By.cssSelector(".price-wrapper");
+
+    public WomenTopsPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Step("Filter items by color")
+    public void filterItemsByColor(String color) {
+        clickElement(colorFilterLocator);
+        clickElement(By.cssSelector("div.filter-options div[option-label='" + color + "']"));
```

Comment:
it would look nicer if you create a separate private method for each dynamic locator and place those methods right after locators. For example you could have getFilerOption method that takes as an argument  string and returns the locator. In like 27 you would just use it with argument 'color'. Also, please check 'String.format' method https://www.geeksforgeeks.org/java-string-format-method-with-examples/ . You could use it for creating strings instead of using plus sign

Reply:
Got it, thanks